R/methods-Chromatograms.R

```diff
@@ -190,3 +212,52 @@ setMethod("plot", signature = signature("Chromatograms"),
                            xlab = xlab, ylab = ylab, main = main, ...)
               }
           })
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{phenoData}: accesses the phenotypical desccription of the
+#'     samples. Returns an \code{NAnnotatedDataFrame} object.
+setMethod("phenoData", "Chromatograms", function(object) object@phenoData)
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{pData}: accesses the phenotypical description of the
+#'     samples. Returns a \code{data.frame}.
+setMethod("pData", "Chromatograms", function(object) pData(phenoData(object)))
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{pData<-}: replace the phenotype data.
+setReplaceMethod("pData", "Chromatograms", function(object, value) {
+    pData(object@phenoData) <- value
+    if (any(rownames(value) != as.character(1:nrow(value))))
+        colnames(object@.Data) <- rownames(value)
```

Comment:
Why do the rownames of the new `pData` take precedence on the colnames of the current object? If they don't match, I think it would be safer to stop with an error, rather than assuming that `pData`'s rownames are the correct one's.